src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx

```diff
@@ -0,0 +1,133 @@
+import TimeSeriesChart from '../../presentational/TimeSeriesChart'
+import { isNumber } from 'lodash'
+
+export type IntradayHeartRatePreviewState = "Default" | "CompleteDataWithThresholds" | "MissingMidDayDataThresholds" | "PartialDataWithThresholds" | "NoData";
+
+export interface IntradayHeartRateChartProps {
+    previewState?: IntradayHeartRatePreviewState,
+    dataSources: DeviceDataV2Namespace[],
+    aggregationOption: IntradayHeartRateAggregationOption,
+    aggregationIntervalMinutes: number,
+    lineColor?: ColorDefinition,
+    thresholds?: ChartThreshold[],
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface IntradayDataPoint {
+    timestamp: number 
+    date: Date
+    value?: number
+    rawValue?: number
+}
+
+export default function IntradayHeartRateChart(props: IntradayHeartRateChartProps) {
+    const [loading, setLoading] = useState(true);
+    const [data, setData] = useState<IntradayDataPoint[] | undefined>(undefined);
+    const [options, setOptions] = useState<MultiSeriesLineChartOptions | undefined>(undefined);
+    const dateRangeContext = useContext(DateRangeContext);
+    const intervalStart = startOfDay(dateRangeContext?.intervalStart ?? new Date());
+    const intervalEnd = startOfDay(add(intervalStart, { days: 1 }));
+    let layoutContext = useContext(LayoutContext);
+    const lineColor = resolveColor(layoutContext.colorScheme, props.lineColor) || '#bbb';
+
+    function initialize() {
+        if (props.previewState) {
+            var data: IntradayHeartRateData = {};
+            switch (props.previewState) {
+                case "Default":
+                case "CompleteDataWithThresholds":
+                    data = FullDayData;
+                    break;
+                case "PartialDataWithThresholds":
+                    data = HalfDayData;
+                    break;
+                case "MissingMidDayDataThresholds":
+                    data = MissingMidDayData;
+                    break;
+            }
+
+            transformToChartData(data);
+        } else {
+            combinedIntradayHeartRateDataProvider(
+                props.dataSources,
+                intervalStart,
+                intervalEnd,
+                props.aggregationOption,
+                props.aggregationIntervalMinutes
+            ).then((data: IntradayHeartRateData) => {
+                transformToChartData(data);
+            }).catch((error: any) => {
+                console.log("error");
+                console.log(error);
+            });
+        }
+    }
+
+    function transformToChartData(data: IntradayHeartRateData) {
+        const start = intervalStart;
+        const end = intervalEnd;
+        const interval: Interval = { start, end };
+        const options: any = { step: 5 };
+        var expectedIntervals = eachMinuteOfInterval(interval, options);
+        var chartData: any[] = expectedIntervals.map((date) => transformToDataPoint(date, data));
```

Comment:
The `transformToDataPoint(...)` function has an implicit return type of `IntradayDataPoint`, so the `chartData` can be declared as an `IntradayDataPoint[]` instead of `any[]`, or the type can be omitted so that it gets that type implicitly.